pkg/aws/sts/cache.go

```diff
@@ -157,7 +157,7 @@ func (c *credentialsCache) CredentialsForRole(ctx context.Context, identity *Rol
 }
 
 func (i *RoleIdentity) String() string {
-	return fmt.Sprintf("%s:%s:%s", i.Role, i.SessionName, i.ExternalID)
+	return fmt.Sprintf("%s:%s:%s", i.Role.Name, i.SessionName, i.ExternalID)
```

Comment:
I think we should change to using the role's full ARN in the cache, rather than just the `Name` (I know it was name previously but this would avoid any name clashes across accounts).

Reply:
Heh good spot I actually already have this change in my currently unpushed code!